build.rs

```diff
@@ -0,0 +1,116 @@
+use std::path::Path;
+
+const CONFIG: &[u8] = include_bytes!("./configs/env.json");
+
+#[derive(serde::Deserialize)]
+struct Config {
+    download_dir: String,
+    url: String,
+    file_names: Vec<String>,
+    sha256_sums: Vec<String>,
+}
+
+fn main() {
+    let config: Config = serde_json::from_slice(CONFIG).expect("Failed to parse config file");
+    download_executables(&config);
+    give_execute_permissions(&config);
+    clone_corelib_repo(&config);
+}
+
+fn download_executables(config: &Config) {
+    let download_dir = format!("{}/{}", env!("HOME"), config.download_dir);
+    if !std::path::Path::new(&download_dir).exists() {
+        std::fs::create_dir_all(&download_dir).expect("Failed to create download directory");
+    }
+    let all_files_exist = config.file_names.iter().all(|filename| {
+        let file_path = format!("{}/{}", download_dir, filename);
+        std::path::Path::new(&file_path).exists()
+    });
+    if all_files_exist {
+        return;
+    }
+    let tar_file_path = format!("{}/starknet-adapter-cli-linux-x86-64.tar.gz", download_dir);
+    let response = reqwest::blocking::get(&config.url).expect("Failed to download file");
+    let mut file = std::fs::File::create(&tar_file_path).expect("Failed to create file");
+    std::io::copy(
+        &mut response.bytes().expect("Failed to read response").as_ref(),
+        &mut file,
+    )
+    .expect("Failed to write to file");
+    let tar_gz = std::fs::File::open(&tar_file_path).expect("Failed to open tar.gz file");
+    let tar = flate2::read::GzDecoder::new(tar_gz);
+    let mut archive = tar::Archive::new(tar);
+    archive
+        .unpack(&download_dir)
+        .expect("Failed to unpack tar.gz file");
+
+    // Validate the unpacked files
```

Comment:
Maybe we should also check that there are *only* the expected files.

Reply:
Done